src/temp/ProjectBadge/ProjectBadge.tsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import NewInfoIcon from '../icons/NewInfoIcon';
+import OffSiteReviewedIcon from '../icons/OffSiteReviewedIcon';
+import FieldReviewedIcon from '../icons/FieldReviewedIcon';
+import styles from './Badge.module.scss';
+import TopProjectIcon from '../icons/TopProjectIcon';
+import { useTranslation } from 'next-i18next';
+
+interface Props {
+  isApproved: boolean;
+  isTopProject: boolean;
+  allowDonations: boolean;
+}
+
+const ProjectBadge = ({ isApproved, isTopProject, allowDonations }: Props) => {
+  const { t } = useTranslation(['common', 'projectDetails']);
+  const getTitleAndIcon = (
+    isApproved: boolean,
+    isTopProject: boolean,
+    allowDonations: boolean
+  ) => {
+    if (!allowDonations)
+      return {
+        icon: <NewInfoIcon width={10} height={10} color={'var(--light)'} />,
+        title: t('common:notDonatable'),
+      };
+
+    if (isTopProject && isApproved)
+      return {
+        icon: <TopProjectIcon color={'var(--light)'} width={11} />,
+        title: t('common:topProject'),
+      };
+    else if (isTopProject && !isApproved)
+      return {
+        icon: <FieldReviewedIcon width={10} color={'var(--light)'} />,
+        title: t('projectDetails:fieldReviewed'),
+      };
+    else if (!isTopProject && !isApproved)
+      return {
+        icon: <OffSiteReviewedIcon width={10} color={'var(--light)'} />,
+        title: t('projectDetails:offSiteReviewed'),
+      };
+  };
```

Comment:
It's a good practice to specify the return type of a function as well, in order to catch unexpected return types.

In this case, the function can return `undefined`. However, we're still rendering empty divs for project badge, badge icon, and badge title in that case. This does not make a difference while rendering the content, but it's still not advisable.

Reply:
This will get called twice for every project in the list, at every render. I'd suggest using useMemo for the calculation, as this logic will not change for a prohect after it is rendered.